---
title: "Thesis project - SARS-CoV-2 UN_48H"
output: html_notebook
---

### Clean environment
```{r}
rm(list=ls())

```
### Libraries
```{r}
library("xlsx") #Read/Write excel files
library(ggrepel) #plots
```
### set directory
```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("C:\\Users\\beasav\\OneDrive - KI.SE\\Bioinformatics\\1. Our data\\SARS-CoV-2")) 
```

### Load data
SARS-CoV-2 UN_48
```{r}
raw_data <- read.xlsx("covid-19_proteomics_UNINF_vs_H48PI_DGE_results.xlsx",1)

```

### Check for missing values in our data

How to do this????
The file does not show any NA value

### Build PCA plot

https://www.nonlinear.com/progenesis/qi-for-proteomics/v4.2/faq/pca.aspx


Flora's code: https://github.com/neogilab/COVID_IFN/blob/main/Proteomics_MERS_v1.Rmd



Does not make sense for only significant values.
```{r}


pca_data <- data.frame(group = condition, t(as.matrix(data_norm)))

```

## Building Vulcano plot
### Filter significantly regulted genes

```{r}
data <- raw_data

data$color <- NA #create a Color column

for (i in 1:nrow(data)){
  if(data[i, 4] > 0.3 & data[i, 8] < 0.05){
    data[i, 10] <- "up_regulated"
  }
  else if(data[i, 4] < -0.3 & data[i, 8] < 0.05){
    data[i, 10] <- "down_regulated"
  }
  else if(data[i, 8] > 0.05){
    data[i, 10] <- "NS" #Non significant
    data[i, 3] <- NA #Delete gene name for NS - So that it doesn't appear in the figure?
  }
  else{ #Label remaining significant genes
    data[i,10] <- "significant"
    data[i, 3] <- NA #Delete gene name for significant - So that it doesn't appear in the figure?
  }
    
}
```

### Volcano plot ggplot

```{r}

pdf("volcano_plot.pdf", height = 5, width = 7)

vol <- ggplot(data, aes(x = logFC, y = -log10(adj.P.Val), colour = logFC)) + geom_point(size=1, alpha = 3, na.rm = T) + scale_color_gradientn(colors = c("#003300", "#325b32","#668466","#ffffcc","#670000")) + theme(legend.title=element_text(size=10),legend.text=element_text(size=6),
        legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
        axis.title.y=element_text(size=10),axis.title.x=element_text(size=10),
        axis.text.y=element_text(size=10),axis.text.x=element_text(size=10)) + labs(x="Log2 Fold Change", y="-log10 (adj.P.Val)") + xlim(-2.5, 2.5) + geom_text_repel(aes(label = Gene_name), size=2.5, color = "black")

vol

dev.off()
```

## Enrichr
### Filter for significant changes

```{r}
sig_Gene_names <- raw_data$Gene_name[data$color != "NS"] #Gene names are stored in raw data; data$color indicates significance
```

### File with dif_expressed Gene names for Enrichr
There's one "Uncharacterized protein" - A0A1W2PPE3 https://www.uniprot.org/uniprot/A0A1W2PPE3.
Since there is no information about the gene it corresponds to, it is not included for Gene Enrichment Analysis

```{r}
write(sig_Gene_names, "dif_expressed genes.txt")

```

### Import Enrichr results

https://maayanlab.cloud/Enrichr/enrich?dataset=b1d4e7523b8b3201144d0f5f8a17f874

Library: BioPlanet 2019

"The p-value is computed from the Fisher exact test which is a proportion test that assumes a binomial distribution and independence for probability of any gene belonging to any set."
*The smaller the p-value, the higher the chance that the genes belong to that pathway, right?*

```{r}
enrichr <- read.delim("Enrichr_BioPlanet_2019_table.txt")

```



#Build a heatmap from our data
BiocManager::install("ComplexHeatmap")

It makes no sense to build a heatmap from only UN_48h, right??
```{r}


```


